v3/src/components/graph/models/graph-data-configuration-model.ts

```diff
@@ -397,27 +397,30 @@ export const GraphDataConfigurationModel = DataConfigurationModel
         }
       })
     },
-    binDetails() {
+    binDetails(_binAlignment?: number, _binWidth?: number) {
       const kDefaultNumberOfBins = 4
       const minValue = self.caseDataArray.reduce((min, aCaseData) => {
         return Math.min(min, self.dataset?.getNumeric(aCaseData.caseID, self.primaryAttributeID) ?? min)
-      }, 0) ?? 0
+      }, Infinity)
       const maxValue = self.caseDataArray.reduce((max, aCaseData) => {
         return Math.max(max, self.dataset?.getNumeric(aCaseData.caseID, self.primaryAttributeID) ?? max)
-      }, 0) ?? 0
+      }, -Infinity)
       const binRange = (maxValue - minValue) / kDefaultNumberOfBins
```

Comment:
I had some trouble with this part and very probably didn't nail it with my solution. I'm pretty sure though that the initial value of min can't be 0 or any other finite number since we don't know what the minimum value is until we iterate over the cases. If we set the initial value to 0 and the minimum value in `caseDataArray` is 3, for example,  we'd end up getting 0 as the minimum value instead of 3.

The latest changes are an improvement, I _think_. Instead of using `Infinity` and `-Infinity`, I initialize both min and max as `NaN` which is at least a little clearer? I believe with using `NaN` here you'd get the same results as with `Infinity` when there are no cases -- both `binWidth` and `binAlignment` would be set to `NaN`. I'm not 100% sure because I can't seem to test that. When there are no cases, you don't get the option to group points into bins. So maybe it's a moot point?

The latest version is a bit more verbose. Assuming it's a definite improvement, I'm wondering if it'd make sense to make one or more separate view methods for getting the min and max numeric values in `caseDataArray`? I'm not sure if that would be useful anywhere else, but it would help keep `binDetails` a little more concise. Do you think that'd be worth doing in this instance? Or is it OK as is?

Reply:
I did not mean to suggest that there's necessarily anything wrong with using `Infinity` in the way you were using it before. I merely meant that it would be bad if those min/max values propagated through and resulted in an attempt to render an axis with range `(Infinity, -Infinity)`, so there needs to be a check somewhere in the code and possibly a decision about what to render in that case. The fact that you said you can't get here when there's no cases goes a long way toward addressing the issue. Did you try grouping points into bins and then deleting the existing cases? In any case, I think the previous version with the infinities is probably clearer than the NaN version, as long as it doesn't propagate out to users.